### ISSUE TITLE
Truffle v5.0.0-beta.2 compilers config error

### ISSUE THREAD

**Issue by jerryji** [trufflesuite/truffle#1452]
- [ ] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue
On Truffle 5.0.0-beta.2 release page ( https://github.com/trufflesuite/truffle/releases/tag/v5.0.0-beta.2 ), "Bring Your Own Compiler" links to https://github.com/trufflesuite/truffle/releases/tag/v5.0.0-beta.0#bring-your-own-compiler , which states "compilers" section config as --

```
  compilers: {
     solc: {
       version: <string>  // ex:  "0.4.20". (Default: Truffle's installed solc)
     }
  }
```

However, "optimizer" does not work _within_ the "compilers" section --

```
do-nyc3-0:~/tmp/ethwalldata$ tail -10 truffle-config.js
    compilers: {
        solc: {
            version: "0.5.0",
            optimizer: {
                enabled: true,
                runs: 1000
            }
        }
    }
};
do-nyc3-0:~/tmp/ethwalldata$ truffle compile --all
Compiling ./contracts/EthWallData.sol...
...
Writing artifacts to ./build/contracts

do-nyc3-0:~/tmp/ethwalldata$ grep \"bytecode\" build/contracts/* | awk '{print $1 " " length($3)/2}' | head -1
build/contracts/EthWallData.json: 25941.5
```

while after removing "compilers" block works (notice the smaller bytecode) --

```
do-nyc3-0:~/tmp/ethwalldata$ tail -10 truffle-config.js
        }
    },
    solc: {
        version: "0.5.0",
        optimizer: {
            enabled: true,
            runs: 1000
        }
    }
};
do-nyc3-0:~/tmp/ethwalldata$ truffle compile --all
Compiling ./contracts/EthWallData.sol...
...
Writing artifacts to ./build/contracts

do-nyc3-0:~/tmp/ethwalldata$ grep \"bytecode\" build/contracts/* | awk '{print $1 " " length($3)/2}' | head -1
build/contracts/EthWallData.json: 16238.5
```

What the issue is, in broad strokes.

## Steps to Reproduce

unzip the attached zip file, compile; update truffle-config.js and recompile to see the difference
[ethwalldata.zip](https://github.com/trufflesuite/truffle/files/2606958/ethwalldata.zip)

Please provide the shortest amount of steps to reproduce your issue.

## Expected Behavior

please refer to problem description above

## Actual Results

please refer to problem description above

## Environment

* Operating System: Ubuntu 18.04 
* Ethereum client: 
* Truffle version (`truffle version`): Truffle v5.0.0-beta.2 - a development framework for Ethereum
* node version (`node --version`): v8.12.0
* npm version (`npm --version`):  6.4.1

**Comment by gnidan** [comment]
This is done and released in v5.0.0. Let us know if you run into it again! Thanks!

**Comment by Manasse228** [comment]
Hello i have an error with this version of truffle, when i type **truffle migrate --network networkName** i have nothing on console. The migrate send nothing like address on console. How to do a good migrate with this new version of truffle ? please need help
